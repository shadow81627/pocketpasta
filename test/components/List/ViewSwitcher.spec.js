import Component from '@/components/List/ViewSwitcher';
import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import BootstrapVuePlugin from 'bootstrap-vue';
import Vue from 'vue';

const localVue = createLocalVue();
let vuetify;

localVue.use(Vuex);
localVue.use(BootstrapVuePlugin);
Vue.use(Vuetify);

describe('ViewSwitcher', () => {
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const factory = () =>
    shallowMount(Component, {
      localVue,
      vuetify,
      stubs: {
        NuxtLink: RouterLinkStub,
      },
    });

  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('currentLayout', () => {
    const wrapper = factory();
    expect(wrapper.vm.currentLayout).toBeTruthy();
  });

  test('availableLayouts', () => {
    const wrapper = factory();
    expect(wrapper.vm.availableLayouts).toBeTruthy();
  });

  test('inputListeners', () => {
    const wrapper = factory();
    expect(wrapper.vm.inputListeners).toBeTruthy();
  });
});
