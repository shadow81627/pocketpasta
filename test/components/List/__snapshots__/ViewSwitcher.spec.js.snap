// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewSwitcher renders properly 1`] = `
"<v-menu-stub opendelay=\\"0\\" closedelay=\\"0\\" contentclass=\\"\\" left=\\"true\\" maxwidth=\\"auto\\" nudgebottom=\\"0\\" nudgeleft=\\"0\\" nudgeright=\\"0\\" nudgetop=\\"0\\" nudgewidth=\\"0\\" openonclick=\\"true\\" closeonclick=\\"true\\" closeoncontentclick=\\"true\\" maxheight=\\"auto\\" offsety=\\"true\\" origin=\\"top left\\" transition=\\"v-menu-transition\\">
  <v-card-stub loaderheight=\\"4\\" tag=\\"div\\">
    <v-list-stub tag=\\"div\\" dense=\\"true\\">
      <v-list-item-stub activeclass=\\"\\" tag=\\"div\\">
        <v-list-item-content-stub>
          <v-list-item-title-stub>Grid</v-list-item-title-stub>
        </v-list-item-content-stub>
        <v-list-item-action-stub>
          <v-icon-stub>$view_comfy</v-icon-stub>
        </v-list-item-action-stub>
      </v-list-item-stub>
    </v-list-stub>
  </v-card-stub>
</v-menu-stub>"
`;
